dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -393,9 +394,23 @@ public Workflow setupWorkflowVersions(String repositoryId, Workflow workflow, Op
         // For each branch (reference) found, create a workflow version and find the associated descriptor files
         for (Triple<String, Date, String> ref : references) {
             if (ref != null) {
-                WorkflowVersion version = setupWorkflowVersionsHelper(workflow, ref, existingWorkflow, existingDefaults,
-                    repository, null, versionName);
-                if (version != null) {
+                WorkflowVersion existingVersion = existingDefaults.get(ref.getLeft());
```

Comment:
Interesting, if I understand correctly. 
We already have the map of references -> commit ids. 
This code is just not creating source file objects when the commit id matches. 

I see the WorkflowVersion gets a magic SKIP_COMMIT_ID which is the only thing that distinguishes a fake version from a real version. We should add a constraint to make sure it doesn't get into the database accidentally or maybe just make a new subclass of WorkflowVersion which dies horribly when we try to write it to a DB.

Reply:
I'm open for discussions, I figured the SKIP_COMMIT_ID probably needs some extra handling to avoid situations like you described.

Are you thinking something like a constraint where commitId can't equal SKIP_COMMIT_ID? I think I'd rather a constraint than a subclass.